BaseCollector.py

```diff
@@ -0,0 +1,8 @@
+from abc import ABC, abstractmethod
+
+
+class BaseCollector(ABC):

```

Comment:
You might add some logging to the BaseCollector. Like argument XYZ is missing etc.
Have a look at the vrops-exporter: https://github.com/sapcc/vrops-exporter/blob/master/BaseCollector.py

Reply:
okay, I will cleanup code and move shared code from the collectors to the baseCollector. Will also streamline configuration class...